### PR TITLE
TGBOT-14-Create_User_repository_&_Bean_using_Map,_CRUD_and_Tests

### PR DESCRIPTION
**Pull request by lamaconda**
got block in tests

### CODE REVIEW

File: stoservice/src/main/java/com/tutrit/stoservice/repository/UserRepository.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public User updateUser(String id, User user) {
+        return userMap.replace(id, user);
+    }
+
+    public User deleteUser(String id) {
+        return userMap.remove(id);
+    }
+}
```
Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/UserRepository.java:28]: Add a counter method that will assign an id for the map

Comment (schneidervs) [inline stoservice/src/main/java/com/tutrit/stoservice/repository/UserRepository.java:28]: public class IngeneerRepository {
private Map<String, Ingeneer> ingeneers = new HashMap<>();

    public Ingeneer createIngeneer(Ingeneer ingeneer);
    public Ingeneer findIngeneer(Ingeneer ingeneer);
    public Ingeneer findIngeneerById(String id);
    public Ingeneer updateIngeneer(Ingeneer ingeneer);
    public Ingeneer deleteIngeneer(Ingeneer ingeneer);
    public Ingeneer deleteIngeneer(String id);
    public int count();
}

File: stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    @Test
+    void updateUser() {
+        userRepository.updateUser("2", new User("2", "name6", "phone6"));
+        assertTrue(UserRepository.userMap.containsKey("2"));
+    }
+
+    @Test
+    void deleteUser() {
```
Comment (schneidervs) [inline stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java:42]: you don't clear the map before each test annotation call. Therefore, you have additional objects from previous tests.

**Comment by schneidervs** [comment]
If the task is still in progress, then you can add WIP to the title

**Review by schneidervs — APPROVED** [review]

**Review by yurasava — DISMISSED** [review]
Not all methods are present.
You probably didn't know, but Max gave a code example with a list of methods and return values here:
https://github.com/Tutritp01/TgBot/pull/13/files
in UserRepository.java

File: stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java (hunk 2 above)
Comment (schneidervs) [reply stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java:42]: > > you don't clear the map before each test annotation call. Therefore, you have additional objects from previous tests.
> 
> it's not necessary

its test separately passes, but batch run fails

File: stoservice/src/main/java/com/tutrit/stoservice/bean/User.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package com.tutrit.stoservice.bean;
+
+import java.util.Objects;
+
+public class User {
+    String id;
+    String name;
+    String phoneNumber;
```
Comment (AlexanderYarokhovich) [inline stoservice/src/main/java/com/tutrit/stoservice/bean/User.java:8]: I would add access modifiers

File: stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        UserRepository.userMap.put("4", new User("4", "user4", "phone4"));
+    }
+
+    @Test
+    void createUser() {
+        userRepository.createUser("5", new User("5", "Boba", "141235"));
+        assertEquals(1, UserRepository.userMap.size());
+    }
+
+}
```
Comment (AlexanderYarokhovich) [inline stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java:47]: I would add an empty line

**Review by AlexanderYarokhovich — APPROVED** [review]

File: stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java (hunk 2 above)
Comment (max-java) [reply stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java:42]: use 
```
    @AfterEach
    void tearDown() {
        UserRepository.userMap = new HashMap<>();
    }
```

File: stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class UserRepositoryTest {
+
+    UserRepository userRepository;
+
+    @BeforeEach
+    void SetUp() {
```
Comment (max-java) [inline stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java:14]: method name should started from lower case

**Review by max-java — CHANGES_REQUESTED** [review]

File: stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java (hunk 5 above)
Comment (lamaconda) [reply stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java:14]: oops)

File: stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java (hunk 4 above)
Comment (lamaconda) [reply stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java:47]: done

File: stoservice/src/main/java/com/tutrit/stoservice/repository/UserRepository.java (hunk 1 above)
Comment (lamaconda) [reply stoservice/src/main/java/com/tutrit/stoservice/repository/UserRepository.java:28]: done

File: stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java (hunk 2 above)
Comment (lamaconda) [reply stoservice/src/test/java/com/tutrit/stoservice/repository/UserRepositoryTest.java:42]: done

**Review by AlexanderYarokhovich — APPROVED** [review]
good

**Review by yurasava — APPROVED** [review]

**Review by Sav4ykVladzislau — APPROVED** [review]

**Review by schneidervs — APPROVED** [review]

**Pull request merged**